### PR TITLE
No poll if success

### PR DESCRIPTION
**Pull request by lmazuel**
Follow more closely Azure guidelines: if the initial response body has a status and the polling strategy was Operation-Location, follow that status.

### CODE REVIEW

File: sdk/core/azure-core/azure/core/polling/base_polling.py [hunk 1]
```diff
@@ -246,6 +246,12 @@ def set_initial_status(self, pipeline_response: "PipelineResponseType") -> str:
 
         self._set_async_url_if_present(response)
 
+        # Check if we can extract status from initial response, if present
+        try:
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/polling/base_polling.py:250]: Should we do this in 

if response.status_code in {200, 201, 202, 204} and self._async_url:

?

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/polling/base_polling.py:250]: Fair feedback

**Review by msyyc — APPROVED** [review]

**Review by kashifkhan — APPROVED** [review]

File: sdk/core/azure-core/tests/async_tests/test_base_polling_async.py [hunk 2]
```diff
@@ -283,6 +283,36 @@ async def send(request, **kwargs):
...
+    # The initial response contains both Location and Operation-Location, a 202 and a success body
+    initial_response = TestBasePolling.mock_send(
+        http_request,
+        http_response,
+        "POST",
+        202,
+        {
+            "operation-location": "http://example.org/async_monitor",
+        },
+        {"status": "succeeded"},
```
Comment (johanste) [inline sdk/core/azure-core/tests/async_tests/test_base_polling_async.py:303]: Let's add some tests for failure terminal states as well....

Comment (xiangyan99) [reply sdk/core/azure-core/tests/async_tests/test_base_polling_async.py:303]: Added.

**Comment by xiangyan99** [comment]
The failure has nothing to do with the change.

**Pull request merged**